### DISCUSSION TITLE
Can GCNConv handle multi graphs?

### DISCUSSION THREAD

**Discussion by Laknath1996** [pyg-team/pytorch_geometric#2725] [Q&A]
Does GCNConv or ChebConv operations support multi-dimensional edge weights? If not, can someone kindly let me know whether there are Graph Conv operators that support multiple edge attributes? The idea is to (if possible) use a GCNConv to explore features of a multi-graph (a graph where a given two nodes can have multiple undirected edges).

**Comment by wsad1** [comment]
- `GCNConv` and `ChebConv` don't support multi-dimensional edge attributes. But you could modify the message function in `GCNConv` to support that .

```
def message(self, x_j, edge_attr):
    # perform some operation like cat, so message depends on node and edge attributes.
    x_j=torch.cat([x_j, edge_attr] 
    return self.linear(x_j) # linear transform to out_channels. 
```

- Yes, there are layers like [NNConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.NNConv), [PNAConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.PNAConv), [GINEConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GINEConv).. that support multiple edge attributes.

**Reply by rusty1s** [reply]
To follow-up on @wsad1, you have two more options:
1. You can simply represent your multi-graph by duplicating edges in `edge_index`.
2. You can iteratively compute the GNN output for each individual edge weight, e.g.:
```python
out = 0
for i in range(num_edge_weights):
    out = out + self.convs[i](x, edge_index, edge_weight[:, i])
```

**Reply by Laknath1996** [reply]
Thank you very much @rusty1s and @wsad1! I'll try out the options you have mentioned above
